### PR TITLE
feat(core): emit event for native search

### LINKED ISSUE #7375: feat(core): input group type search emit event when native clear button clicked
#### Is this a bug, enhancement, or feature request?

Enhancement.

#### Briefly describe your proposal.

It's possible to set `type='search'` for input group, in that case some browsers add clear button. Add event that are emitted when clear button is pressed.

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

0.33.0-rc.214

### PR DESCRIPTION
**Pull request by StevenZhao318**
## Related Issue(s)

closes #7375 

## Description

Emit an event for native input search events. 

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [n/a] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [n/a] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/input-group/input-group.component.ts [hunk 1]
```diff
@@ -145,6 +145,10 @@ export class InputGroupComponent implements ControlValueAccessor, OnInit, OnDest
     @Output()
     addOnButtonClicked: EventEmitter<any> = new EventEmitter<any>();
 
+    /** Event emitted when the native clear button is clicked. or when native search is executed */
+    @Output()
+    search: EventEmitter<any> = new EventEmitter<any>();
```
Comment (dmitry-stepanenko) [inline libs/core/src/lib/input-group/input-group.component.ts:150]: please use proper event type

File: libs/core/src/lib/input-group/input-group.component.ts [hunk 2]
```diff
@@ -262,6 +266,11 @@ export class InputGroupComponent implements ControlValueAccessor, OnInit, OnDest
         this.addOnButtonClicked.emit($event);
     }
 
+    /** @hidden */
+    onSearchEvent($event: Event) {
```
Comment (dmitry-stepanenko) [inline libs/core/src/lib/input-group/input-group.component.ts:270]: linter will complain for the absence of return type
```suggestion
    onSearchEvent(event: Event): void {
```

File: libs/core/src/lib/input-group/input-group.component.ts (hunk 1 above)
Comment (mikerodonnell89) [reply libs/core/src/lib/input-group/input-group.component.ts:150]:
```suggestion
    search: EventEmitter<Event> = new EventEmitter<Event>();
```

**Review by dmitry-stepanenko — APPROVED** [review]

**Comment by droshev** [comment]
@StevenZhao318 your commit messages need to start with `fix(....):`, `feat(...):`, `test(...):`,... `libs(..):` is not one of those we use. That's why CI is failing.

**Review by mikerodonnell89 — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Pull request merged**